finetune/adapter.py

```diff
@@ -229,6 +231,7 @@ def fit(
             if fabric.global_rank == 0:
                 copy_config_files(checkpoint_dir, checkpoint_file.parent)
                 save_hyperparameters(setup, checkpoint_file.parent)
+                save_prompt_style(data.prompt_style, checkpoint_file.parent)
```

Comment:
The block of code above could probably move to a function. Or even a checkpointer. More like a checkpoint callback in, like in ...

<details>
<summary></summary>

<details>
<summary></summary>

<details>
<summary></summary>

<details>
<summary></summary>

<details>
<summary></summary>

**T     R     A     I     N     E     R**

</details>

</details>

</details>

</details>

</details>